File: processes_learning/elixir/my_bot_1.ex
defmodule MyBot do
  @moduledoc"""
    iex> pid = spawn(MyBot, :init, [1])
    iex> send(pid, :start)
    iex> Process.alive?(pid)
  """
  def init(bot_number) do
    receive do
      :say_hello->
        IO.puts "🤖 Bot <#{bot_number}> :: Hello! ✌️..."
      :say_bye ->
        IO.puts "🤖 Bot <#{bot_number}> :: Bye! 👋"
    end

    init(bot_number)
  end
end

File: processes_learning/erlang/my_bot_1.erl
-module(my_bot_1).
-export([init/1]).

init(Bot_number) ->
  receive
    say_hello ->
      io:format(">> Bot ~p :: Hello! ✌️\n", [Bot_number]);

    say_bye ->
      io:format(">> Bot ~p :: Bye! ✌️\n", [Bot_number])
  end,
  init(Bot_number).

%%% How to run this code?
%%% Pid = spawn(my_bot_1, init, [1]).
%%% Pid ! say_hello.
%%% is_process_alive(Pid).
%%% Pid ! say_bye.
